packages/nimble-components/src/wafer-map/specs/README.md

```diff
@@ -0,0 +1,199 @@
+
+### Blazor integration
+
+Blazor support is out of scope for the current implementation.
+
+### Visual Appearance
+
+For the initial component, we are going with the original wafer map appearance. 
+![WaferMap Example](file:///Resources/wafer_map_example.png)
+
+## Implementation
+
+As mentioned above, this is not a component that we start from scratch, but a visualization that already exists which is outdated and not futureproof. 
+As a first step, we are working on the "bloat removal" which is basically the removal of unwanted but exiting features (legend, tooltip etc.) and dependencies.
+
+Original component: [wafermap](https://dev.azure.com/ni/DevCentral/_git/op-web-portal?path=/libraries/wafermap&version=GBfeature/microServicesSplit)\
+Once the "bloat removal" is finished, the next step is to adapt the current 
+
+### Code submission strategy
+
+In order to make the process of submitting the re-written wafer map component easier for us and for the code reviewers as well, we propose the following bottom-up approach:
+* As an initial step, we submit a skeleton of the `nimble-wafer-map` component into the main branch. This will serve as a starting point on which the complete functionality will be built on.
+* Once the skeleton is in place, we start to work on and submit individual pieces of the functionality so it's not very overwhelming to review the code.
+
+### States
+
+We are not planning to allow the user to generate an “empty" wafer map, so we must ensure that data is provided when instantiating a new `nimble-wafer-map` component.
+The only action which we could consider as a "state transition" is the zoom functionality which was specified in the sections above.
+We expect that the end user be able to zoom in on the dies withing the wafer map and be able to zoom out by using the mouse wheel.
+Also, the user must be allowed to pan the zoomed in wafer map into any direction of its XY axis by holding the left mouse button and moving the mouse cursor around.
+
+Another interaction, which must be part of the component is the ability to identify the location of the mouse pointer on the wafer map die area.
+
+### Globalization
+
+Currently there will be no string data presented by this component. Might change in the future, but for now it will contain only numerical data.
+
+### Security
+
+N/A
+
+### Performance
+
+As mentioned in the "Risks" section, the biggest challenge from the performance standpoint is the representation of large number of datapoints and interacting with these datapoints in the browser.
+Currently the only thing we would like to achieve is to measure (benchmark) the time it takes a `nimble-wafer-map` component to be loaded with various number of datapoints.
+
+### Dependencies
+
+The only 3rd party dependency on which the current wafer map component depends on is the [D3(Data-Driven Documents)](https://d3js.org/) library.
+We use parts of this library in our zooming and rendering functionality.
+
```

Comment:
Do you have a sense for our plan for updating the version of d3 going forward?
1. how often do they publish updates? Are they usually pretty stable or lots of breaking changes?
2. how often do they have security vulnerabilities (directly or in their sub-dependencies) and how quickly do they fix them?

I ask because we don't have automated dependency updates turned on for this repo yet because we have so few third-party dependencies. If you think it would be valuable to update d3 frequently, we could prioritize the work to update it automatically.

Reply:
We use pretty much the basic functionality from the d3 library. 
I verified they repo and the last breaking change happened with the release of 7.0.0 release. (June 2021).
From the release history the release pattern is the following:
1. Major version change seems to happen once a year
2. Minor version change happens ~every three months
3. Revision changes are random - sometimes revision changes in every few days, weeks

We don't expect to be aligned with the latest version all time, however, we expect not to miss important fixes regarding modules we use or security updates.

Reply:
Cool. We have basic security auditing on our pipeline so we should get notified of vulnerabilities and that will trigger us to look for security updates.

Longer term as part of #291 I would like to turn on Renovate or Dependabot in this repo to automatically create PRs to update deps on some cadence. But given the versioning you described I don't think it will be too burdensome to keep our versions pinned most of the time and to update them manually when needed.